arithmetization/src/test/java/net/consensys/linea/zktracer/exceptions/OutOfGasMemExpExceptionTest.java

```diff
@@ -0,0 +1,604 @@
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, 0, 1})
+  void outOfGasExceptionCodeCopy(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        .push(Bytes.fromHexString("0xFA")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(3) // size
+        .push(0) // offset
+        .push(32) // destoffset
+        .op(OpCode.CODECOPY); // Should copy 60fa60 (first 3 bytes of the code)
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost();
+
+    bytecodeRunner.run(gasCost + cornerCase);
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, 0, 1})
+  void outOfGasExceptionExtCodeCopy(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        // constructor
+        .push(
+            Bytes.fromHexString(
+                "0x7FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(
+            Bytes.fromHexString(
+                "0xFF60005260206000F30000000000000000000000000000000000000000000000")) // value
+        .push(32) // offset
+        .op(OpCode.MSTORE)
+        // Create the contract
+        .push(41)
+        .push(0)
+        .push(0)
+        .op(OpCode.CREATE)
+        .push(32) // size
+        .push(0) // offset
+        .push(33) // destoffset
+        .op(OpCode.DUP4) //
+        .op(OpCode.EXTCODECOPY);
+
+    Bytes pgCompile = program.compile();
+    BytecodeRunner bytecodeRunner = BytecodeRunner.of(pgCompile);
+
+    long gasCost = bytecodeRunner.runOnlyForGasCost();
+
+    bytecodeRunner.run(gasCost + cornerCase);
+    if (cornerCase == -1) {
+      assertEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    } else {
+      assertNotEquals(
+          OUT_OF_GAS_EXCEPTION,
+          bytecodeRunner.getHub().previousTraceSection().commonValues.tracedException());
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {-1, 0, 1})
+  void outOfGasExceptionReturn(int cornerCase) {
+    BytecodeCompiler program = BytecodeCompiler.newProgram();
+
+    program
+        .push(
+            Bytes.fromHexString(
+                "0xF00000000000000000000000000000000000000000000000000000000000FFAA")) // value
+        .push(0) // offset
+        .op(OpCode.MSTORE)
+        .push(3) // value
```

Comment:
`value` is really a `size`

Reply:
There should be a second version of this test where the transaction is a `DEPLOYMENT_TRANSACTION`. The pricing of the `RETURN` is different in this case (you pay 200 gas for every byte you are attempting to deploy.) The `program` you provide would have to be passed as the `Transaction.payload` instead. We have several tests where we use deployment transactions.